patcher/src/ts/sidebar/ServerSelect.tsx

```diff
@@ -49,66 +54,171 @@ class ActiveServerView extends React.Component<ActiveServerViewProps, ActiveServ
+
+    events.fire('play-sound', 'select');
+
+    //is there a required order here?
+    dispatch(changeChannel(server.channelInfo));
+    dispatch(changeServer(server.serverInfo));
+    dispatch(fetchCharacters());
+    dispatch(selectCharacter(null));
   }
 
-  generateActiveView = (server: any) => {
-    return <ActiveServerView item={server} />
+  generateActiveView = (server: IServerOption) => {
+    if(server.serverInfo)
+      return <ActiveServerView item={server.serverInfo} />
+     else
+    {
+      //todo component
+      let status = server.channelInfo.channelStatus == 4 ? 'online' : 'offline'; //more than one status should show online...
+      let statusDesc = ChannelStatus[server.channelInfo.channelStatus];
+
+       return (
+        <div className='server-select quickselect-list'>
+          <div>
+            <div className='server-status'><div className={'indicator ' + status} data-position='right'
+              data-delay='150' data-tooltip='offline' /></div>
+            <div className='server-details'>
+              <h5 className='server'>{server.channelInfo.channelName} ({statusDesc})</h5>
+            </div>
+          </div>
+        </div>
+      );
+    }
   }
 
-  generateListView = (server: any) => {
-    return <ServerListView item={server} />
+  generateListView = (server: IServerOption) => {
+
+    if(server.serverInfo)
+      return <ServerListView item={server} />
+    else
+    {
+      //todo component
+      let status = server.channelInfo.channelStatus == 4 ? 'online' : 'offline'; //more than one status should show online...
+      let statusDesc = ChannelStatus[server.channelInfo.channelStatus];
+
+       return (
+        <div className='server-select quickselect-list'>
+          <div>
+            <div className='server-status'><div className={'indicator ' + status} data-position='right'
+              data-delay='150' data-tooltip='offline' /></div>
+            <div className='server-details'>
+              <h5 className='server'>{server.channelInfo.channelName} ({statusDesc})</h5>
+            </div>
+          </div>
+        </div>
+      );
+    }
   }
 
   getSelectedIndex = () : number => {
-    return this.props.servers.indexOf(this.props.selectedServer);
+    const { currentServer } = this.props.serversState;
+    const { selectedChannel } = this.props.channelsState;
+    if(!currentServer && !selectedChannel)
+      return 0;
+
+    return this.listAsArray.indexOf(this.listAsArray.find(i => {
+      if(i.serverInfo && currentServer)
+        return i.serverInfo.channelID == currentServer.channelID
+      else if(i.channelInfo && selectedChannel)
+        return i.channelInfo.channelID == selectedChannel.channelID
+      
+      return false;
+    }));
+  }
+
+  mergeServerChannelLists(servers:Array<Server>, channels:Array<Channel>) : Array<IServerOption> {
+
+    let filteredServers:Array<Server> = servers;
+    if(patcher.getScreenName().search(/^cse/i) === -1)
```

Comment:
https://github.com/airbnb/javascript#blocks--braces

    if (expression) statement;

or

    if (expression) {
      statement;
    }

are the acceptable forms.